### PR TITLE
source team: add mission to team page

### PR DESCRIPTION
**Pull request by mrnugget**
This is a first draft, a first proposal.

Goal is for us to define what our north star is and how we get there. I grouped both under `## Mission` here and put my personal proposal on.

I expect you to point out where I'm off, to suggest ideas, to correct me where I'm wrong, to add what you want to see as our mission.

So: let's go, comment & suggest! 🙂

### CODE REVIEW

File: content/departments/engineering/teams/source/strategy.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+
+We believe this is critical for building a long-term defensible value for Sourcegraph as all other code-search solutions are focused on searching across and individual repo or code host.
+
+## Mission
+
+The Source teams is responsible for the syncing, storing, and securing of code in Sourcegraph's code AI platform. We are putting the Source in Sourcegraph.
+
+We are building a foundation on which the rest of Sourcegraph stands. No other part of Sourcegraph can work without source code and properly managed access to it.
+
+Source makes Sourcegraph's code AI platform _universal_, by allowing it to access any code host, through any authentication and authorization provider, that customers require.
```
Comment (sashaostrikov) [inline content/departments/engineering/teams/source/strategy.md]: Do you think we should add anything about licensing here? In terms of making it fair and inclusive for every customer, big and small?

File: content/departments/engineering/teams/source/strategy.md [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
-TL;DR: The Source department builds value for Sourcegraph by ensuring customers can index all of their code regardless of code host. We believe this is critical for building a long-term defensible value for Sourcegraph as all other code-search solutions are focused on searching across and individual repo or code host.
+## Strategy
+
+The Source team builds value for Sourcegraph by ensuring customers can index and secure all of their code regardless of code host and auth provider.
+
+We believe this is critical for building a long-term defensible value for Sourcegraph as all other code-search solutions are focused on searching across and individual repo or code host.
+
+## Mission
+
+The Source teams is responsible for the syncing, storing, and securing of code in Sourcegraph's code AI platform. We are putting the Source in Sourcegraph.
```
Comment (sashaostrikov) [inline content/departments/engineering/teams/source/strategy.md]: typo: 
```suggestion
The Source team is responsible for the syncing, storing, and securing of code in Sourcegraph's code AI platform. We are putting the Source in Sourcegraph.
```

File: content/departments/engineering/teams/source/strategy.md [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+
+We are building a foundation on which the rest of Sourcegraph stands. No other part of Sourcegraph can work without source code and properly managed access to it.
+
+Source makes Sourcegraph's code AI platform _universal_, by allowing it to access any code host, through any authentication and authorization provider, that customers require.
+
+We are building a product for site-admins and a platform for engineers at Sourcegraph.
+
+What we build is reliable, predictable, maintainable, debuggable.
+
+Sourcegraph site-admins trust the tools we build for them. They enjoy using them, because they help them get their job done faster and more easily, with less ambiguity.
```
Comment (sashaostrikov) [inline content/departments/engineering/teams/source/strategy.md]: We can add something about smooth and magical onboarding: Sourcegraph can be set up in 3 clicks, and then you forget about it because it just works.

**Review by sashaostrikov — APPROVED** [review]
LGTM! Added a couple of suggestions.

File: content/departments/engineering/teams/source/strategy.md (hunk 3 above)
Comment (erzhtor) [inline content/departments/engineering/teams/source/strategy.md]:
```suggestion
Sourcegraph site-admins trust and enjoy using the tools we build because they can easily manage them. Our goal is to provide a smooth, intuitive product that just works.
```

I like the below phrasing that engineers don't have to know/worry about the underlying source-related things work. I was thinking about putting a similar tone for the admin experience too.

**Review by erzhtor — APPROVED** [review]

File: content/departments/engineering/teams/source/strategy.md [hunk 4]
```diff
@@ -1,3 +1,23 @@
-# Source Department Strategy
+# Source Strategy & Mission
 
-TL;DR: The Source department builds value for Sourcegraph by ensuring customers can index all of their code regardless of code host. We believe this is critical for building a long-term defensible value for Sourcegraph as all other code-search solutions are focused on searching across and individual repo or code host.
+## Strategy
+
+The Source team builds value for Sourcegraph by ensuring customers can index and secure all of their code regardless of code host and auth provider.
+
+We believe this is critical for building a long-term defensible value for Sourcegraph as all other code-search solutions are focused on searching across and individual repo or code host.
```
Comment (kopancek) [inline content/departments/engineering/teams/source/strategy.md]: Should we generalise this a bit, since we do not serve that code to code-search anymore?

File: content/departments/engineering/teams/source/strategy.md (hunk 1 above)
Comment (kopancek) [inline content/departments/engineering/teams/source/strategy.md]: It's nice to say `any authentication and authroization provider, that customers require`, but in reality we need to make hard calls which ones we want to support and which one not. 

Should we maybe reword this to be clear about the expectations here? E.g. something like:
```suggestion
Source makes Sourcegraph's code AI platform _universal_, by allowing it to access any code host, through the majority of authentication and authorization providers that our customers use.
```

Comment (kopancek) [reply content/departments/engineering/teams/source/strategy.md]: Also, since we own Own, should we add something about it here?

File: content/departments/engineering/teams/source/strategy.md [hunk 5]
```diff
@@ -1,3 +1,23 @@
...
+
+## Mission
+
+The Source teams is responsible for the syncing, storing, and securing of code in Sourcegraph's code AI platform. We are putting the Source in Sourcegraph.
+
+We are building a foundation on which the rest of Sourcegraph stands. No other part of Sourcegraph can work without source code and properly managed access to it.
+
+Source makes Sourcegraph's code AI platform _universal_, by allowing it to access any code host, through any authentication and authorization provider, that customers require.
+
+We are building a product for site-admins and a platform for engineers at Sourcegraph.
```
Comment (indradhanush) [inline content/departments/engineering/teams/source/strategy.md:17]:
```suggestion
What we build is reliable, predictable, maintainable, and debuggable.
```

File: content/departments/engineering/teams/source/strategy.md [hunk 6]
```diff
@@ -2,2 +1,23 @@
...
+
```
Comment (indradhanush) [inline content/departments/engineering/teams/source/strategy.md:18]:
```suggestion
Sourcegraph site-admins trust the tools we build for them. They enjoy using them, because they help them get their job done faster and easier, with less ambiguity.
```

File: content/departments/engineering/teams/source/strategy.md [hunk 7]
```diff
@@ -2,2 +1,23 @@
...
+What we build is reliable, predictable, maintainable, debuggable.
+
```
Comment (indradhanush) [inline content/departments/engineering/teams/source/strategy.md:20]:
```suggestion
Engineers at Sourcegraph love to use our APIs because they are reliable and never leak: users of our APIs don't have to worry about how gitserver, repo-updater, authz providers, or anything else works.
```

**Review by indradhanush — APPROVED** [review]
Looks good. I like the direct to the point approach here. Left a bunch of nits. 

I agree with Alex's comment about licensing.

Edit: Deleted my nits because my review tool misplaced them. 🤦 (and added them back manually 🤦 🤦 🤦 )

File: content/departments/engineering/teams/source/strategy.md (hunk 3 above)
Comment (kopancek) [inline content/departments/engineering/teams/source/strategy.md]: I would say that our mission is to provide site-admins with tools that they do not need often, because "it all just works ™️ ". Building on reliable, predictable, etc. above.

File: content/departments/engineering/teams/source/strategy.md [hunk 8]
```diff
@@ -1,3 +1,23 @@
...
+Sourcegraph site-admins trust the tools we build for them. They enjoy using them, because they help them get their job done faster and more easily, with less ambiguity.
+
+Engineers at Sourcegraph love to use our APIs because they are reliable and never leak: users of our APIs don't have to worry about how gitserver, or repo-updater, authz providers, or anything else works.
```
Comment (kopancek) [inline content/departments/engineering/teams/source/strategy.md]: This is an especially nice sentence ❤️

Comment (indradhanush) [inline content/departments/engineering/teams/source/strategy.md]:
```suggestion
Engineers at Sourcegraph love to use our APIs because they are reliable and never leak: users of our APIs don't have to worry about how gitserver, repo-updater, authz providers, or anything else works.
```

File: content/departments/engineering/teams/source/index.md [hunk 9]
```diff
@@ -8,7 +8,9 @@ The Source team owns everything related to
 - gitserver
 - repo-updater
 
-We provide value to Sourcegraph by ensuring Sourcegraph can reliably index code from any code hosts, with the proper authentication and authorization in place. For more details, read our [Source Department strategy](strategy.md)
+We provide value to Sourcegraph by ensuring Sourcegraph can reliably index code from any code hosts, with the proper authentication and authorization in place.
+
+For more details, read our [Source strategy & mission](strategy.md)
```
Comment (indradhanush) [inline content/departments/engineering/teams/source/index.md:13]:
```suggestion
For more details, read about our [strategy & mission](strategy.md).
```

File: content/departments/engineering/teams/source/index.md [hunk 10]
```diff
@@ -8,7 +8,9 @@ The Source team owns everything related to
 - gitserver
 - repo-updater
 
-We provide value to Sourcegraph by ensuring Sourcegraph can reliably index code from any code hosts, with the proper authentication and authorization in place. For more details, read our [Source Department strategy](strategy.md)
+We provide value to Sourcegraph by ensuring Sourcegraph can reliably index code from any code hosts, with the proper authentication and authorization in place.
```
Comment (indradhanush) [inline content/departments/engineering/teams/source/index.md]:
```suggestion
We provide value to Sourcegraph by ensuring Sourcegraph can reliably index code from any code host, with the proper authentication and authorization in place.
```

File: content/departments/engineering/teams/source/strategy.md (hunk 1 above)
Comment (sashaostrikov) [reply content/departments/engineering/teams/source/strategy.md]: oh yeah, I forgot about Code ownership 😁 nice one!

Comment (pjlast) [reply content/departments/engineering/teams/source/strategy.md]: I agree it should be mentioned if we're responsible for it. That being said, throwing even more things in here makes the mission/vision feel quite fractured, and at some point it stops feeling like "here is our focus, here's what we aim to do and do well" and instead feels like "here's everything we're trying to juggle, and we hope to not drop too many balls along the way"

Comment (mrnugget) [reply content/departments/engineering/teams/source/strategy.md]: > That being said, throwing even more things in here makes the mission/vision feel quite fractured

Agree on this: I think the mission should hold regardless of which exact feature we work on. Example: we also don't want to list "we own GitHub App here", right?

File: content/departments/engineering/teams/source/strategy.md (hunk 4 above)
Comment (mrnugget) [reply content/departments/engineering/teams/source/strategy.md]: We do! We are a platform team and Code Search and Cody use our APIs. Code Search team is not going to build their own gitserver and authz provider, right?

Comment (varsanojidan) [inline content/departments/engineering/teams/source/strategy.md]:
```suggestion
We believe this is critical for building a long-term defensible value for Sourcegraph as all other code-search solutions are focused on searching across an individual repo or code host.
```

**Review by varsanojidan — APPROVED** [review]

File: content/departments/engineering/teams/source/index.md (hunk 10 above)
Comment (BolajiOlajide) [inline content/departments/engineering/teams/source/index.md]:
```suggestion
We provide value to Sourcegraph by ensuring Sourcegraph can reliably index code from any code hosts, with the proper authentication and authorization in place. We also ensure administrators are properly empowered to configure Sourcegraph to the best of their requirements by creating an awesome admin experience.
```

Comment (BolajiOlajide) [reply content/departments/engineering/teams/source/index.md]: I feel we should talk a bit about the admin experience here since it's also in our purview.

File: content/departments/engineering/teams/source/strategy.md (hunk 4 above)
Comment (BolajiOlajide) [inline content/departments/engineering/teams/source/strategy.md]:
```suggestion
The Source team builds value for Sourcegraph by ensuring customers can index and secure their code regardless of the code host and auth provider.

This is critical for building a long-term defensible value for Sourcegraph, as all other code-search solutions focus on searching across an individual repo or code host.
```

Just a suggestion to make this more straightforward to understand, In my opinion.

**Review by BolajiOlajide — APPROVED** [review]

File: content/departments/engineering/teams/source/index.md (hunk 10 above)
Comment (eseliger) [reply content/departments/engineering/teams/source/index.md]: I think this is kind of implicit, because if admins cannot configure it right, they don't get code indexed and authorized properly. We also own src-cli and own and so forth, but don't mention them specifically here, I think going into detail on the strat and mission page can be enough. This as-is is a simple one-liner that we can preach every time people ask for our purpose.

Comment (eseliger) [inline content/departments/engineering/teams/source/index.md]:
```suggestion
We provide value to Sourcegraph by ensuring all Sourcegraph features can reliably access code from any code hosts, with the proper authentication and authorization in place.
```

maybe to clearly say our purpose is to serve other teams building the features for our customers, so they don't have to care? Also, it's not just about indexing, it's also reading diffs, listing branches, applying patches, and so forth so "access" feels more generic to me.

File: content/departments/engineering/teams/source/strategy.md (hunk 4 above)
Comment (eseliger) [reply content/departments/engineering/teams/source/strategy.md]: I don't think you two disagree, we should codify better that we serve for all the products Sourcegraph offers.

Comment (eseliger) [inline content/departments/engineering/teams/source/strategy.md]:
```suggestion
The Source team provides value for Sourcegraph by ensuring customers can connect all of their code hosts and auth providers to Sourcegraph products.

We believe this is critical for building a long-term defensible value for Sourcegraph as all other devtool solutions are focused on individual repos or code hosts. We make Sourcegraph cross-platform and support niche products that don't get the attention that single GitHub.com code host companies get.
```

A bit of a mouthful so we might want to iterate on this wording a bit more, but I think it's important to call out that we support _all the products_ we are eventually going to build, and that we provide massive value in making Sourcegraph the cross-code-host tool that is is. Also, supporting niche markets like perforce, gerrit and such makes us stand out heavily because no other company out there is investing in those tools massively, at least I've never seen a Gerrit CIP solution before.

**Review by eseliger — APPROVED** [review]

File: content/departments/engineering/teams/source/index.md (hunk 10 above)
Comment (mrnugget) [reply content/departments/engineering/teams/source/index.md]: Yup, agree with @eseliger here. One liner on the team page is enough I think.

Comment (mrnugget) [reply content/departments/engineering/teams/source/index.md]: d5cf2ce97]

Comment (mrnugget) [reply content/departments/engineering/teams/source/index.md]: d5cf2ce97]

File: content/departments/engineering/teams/source/strategy.md (hunk 4 above)
Comment (mrnugget) [reply content/departments/engineering/teams/source/strategy.md]: Since this is more about general market strategy, I only made a slight adjustment here, but I think it fixes the "problem"

Comment (mrnugget) [reply content/departments/engineering/teams/source/strategy.md]: Good point. I simplified it a bit.

File: content/departments/engineering/teams/source/strategy.md (hunk 1 above)
Comment (mrnugget) [reply content/departments/engineering/teams/source/strategy.md]: I reworded it to remove `any`, which is the critical word here, IMHO.

File: content/departments/engineering/teams/source/strategy.md (hunk 3 above)
Comment (mrnugget) [reply content/departments/engineering/teams/source/strategy.md]: I would say that it's a bit too concrete for a mission statement: I don't want us to nail down the exact "tools" we build for them, because the mission is not to build nice onboarding (that could be _one_ of the things we do to fulfill the mission), but it's to build something that admins trust.

Comment (mrnugget) [reply content/departments/engineering/teams/source/strategy.md]: Incorporated 2nd part, I think the first part I'll leave as is because they shouldn't "manage the tools", they should use the tools to get a job done. I think we're in alignment though and it's just wording.

Comment (mrnugget) [reply content/departments/engineering/teams/source/strategy.md]: Fixed after incorporating @erzhtor's suggestion above.
